kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/ClusterService.java

```diff
@@ -72,4 +95,19 @@
                             .map(c -> ClusterUtil.convertToConsumerGroup(c, cluster)).collect(Collectors.toList()))
                     .map(s -> ResponseEntity.ok(Flux.fromIterable(s)));
     }
+
+    private void fillOffsetParams(ConsumerDetail consumerDetail, Map<TopicPartition, OffsetAndMetadata> topicMetadata) {
+        List<Long> currentOffsets = new ArrayList<>();
+        List<Long> behindMessagesList = new ArrayList<>();
+        var endOffsets = consumerDetail.getEndOffset();
+        var topics = consumerDetail.getTopic();
+        var partitions = consumerDetail.getPartition();
+        for (int i = 0; i < consumerDetail.getTopic().size(); i++) {
```

Comment:
I don't think this code will do you expect. It will add 1 partition per topic to offsets